File: app/src/main/java/org/schulcloud/mobile/data/model/Topic.java
package org.schulcloud.mobile.data.model;

import io.realm.RealmList;
import io.realm.RealmObject;

public class Topic extends RealmObject {
    public String _id;
    public String name;
    public String description;
    public String date;
    public String time;
    public String courseId;
    public Boolean hidden;
    public RealmList<Contents> contents;
}

File: app/src/main/java/org/schulcloud/mobile/models/topic/Topic.kt
package org.schulcloud.mobile.models.topic

import com.google.gson.annotations.SerializedName
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.schulcloud.mobile.models.base.HasId
import org.schulcloud.mobile.models.content.ContentWrapper
import org.schulcloud.mobile.utils.HOST

open class Topic : RealmObject(), HasId {

    @PrimaryKey
    @SerializedName("_id")
    override var id: String = ""

    var courseId: String? = null
    var name: String? = null
    var time: String? = null
    var date: String? = null
    @Suppress("unused")
    var position: Int? = null
    var contents: RealmList<ContentWrapper>? = null

    val url: String
        get() = "$HOST/courses/$courseId/topics/$id"
}
